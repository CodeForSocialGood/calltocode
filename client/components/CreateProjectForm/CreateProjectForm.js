import React, { Component } from 'react'
import { Field, reduxForm } from 'redux-form'

import UploadDropzone from '../UploadDropzone/UploadDropzone'
import styles from './CreateProjectForm.scss'

class CreateProjectForm extends Component {
  renderImageUpload (field) {
    return (
      <UploadDropzone className={styles.inputImageUpload}
        saveFile={file => field.input.onChange(file)} />
    )
  }

  renderProjectName (field) {
    return (
      <input className={styles.inputProjectName}
        placeholder="Project Name"
        title = {field.meta.error}
        {...field.input} />
    )
  }
  render () {
    return (
      <form className={styles.form}>
        <h1 className={styles.title}>Create New Project</h1>

        <Field name="image"
          component={this.renderImageUpload} />

        <Field name="project-name"
          component={this.renderProjectName} />

        <button className={styles.buttonSubmit} type="submit">
          Create Project
        </button>
      </form>
    )
  }
}

const CreateProjectFormRedux = reduxForm({
  form: 'CreateProjectForm',
  onSubmitSuccess: (result, dispatch) => {

  }
})(CreateProjectForm)

export default CreateProjectFormRedux
