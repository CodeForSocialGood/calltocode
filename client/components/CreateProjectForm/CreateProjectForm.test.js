/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {CreateProjectForm} from './CreateProjectForm'
let mockProps

beforeEach(() => {
  mockProps = {
    user: {
      organization: null
    },
    handleSubmit: () => {}
  }
})

describe('<CreateProjectForm />', () => {
  describe('with valid user organization', () => {
    beforeEach(() => { mockProps.user.organization = 'fakeOrg' })

    test('component renders correctly', () => {
      const component = shallow(<CreateProjectForm {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })
})
