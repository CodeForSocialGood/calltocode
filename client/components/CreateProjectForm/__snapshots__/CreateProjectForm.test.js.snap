// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateProjectForm /> with valid user organization component renders correctly 1`] = `
<form
  className="form"
  id="createProjectForm"
>
  <h1
    className="formHeading"
  >
    Create New Project
  </h1>
  <Field
    component={[Function]}
    name="image"
  />
  <Field
    component={[Function]}
    name="projectname"
  />
  <button
    className="buttonSubmit"
    type="submit"
  >
    Create Project
  </button>
</form>
`;
